src/targets/gpu/compile_hip.cpp

```diff
@@ -246,6 +247,16 @@ compile_hip_src(const std::vector<src_file>& srcs, std::string params, const std
             MIGRAPHX_THROW("Missing hsaco");
         };
 
+    if(enabled(MIGRAPHX_GPU_DUMP_SRC{}))
+    {
+        for(const auto& src : srcs)
+        {
+            if(src.path.extension() != ".cpp")
+                continue;
+            std::cout << std::string(src.content.first, src.len()) << std::endl;
```

Comment:
Is there an advantage to doing it this way instead of 
```
if(src.path.extension() == ".cpp")
    std::cout << std::string(src.content.first, src.len()) << std::endl;
```

Reply:
It just reduces the nested scopes.